core/src/main/resources/error/error-classes.json

```diff
@@ -933,6 +933,12 @@
     ],
     "sqlState" : "42604"
   },
+  "INVALID_TYPE_FOR_FILTER_EXPR" : {
```

Comment:
Is it a specific case of `DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE`. Can't you re-use it here?

Reply:
It isn't really a parameter. It's debatable whether it should be a subclass. But DATATYPE_MISMATCH is really turning into a catchall. I would either leave it as is, or make it:
DATATYPE_MISMATCH.FILTER_NOT_BOOLEAN or so..

Reply:
Yes, I agree that it's not really related to an input parameter.

Let me put it in the `DATATYPE_MISMATCH` and rename it as `FILTER_NOT_BOOLEAN` for now.